src/Client.php

```diff
@@ -19,13 +18,23 @@ class Client implements HttpClient
 {
     private $client;
     private $responseFactory;
-    private $streamFactory;
 
-    public function __construct(Artax\Client $client = null, ResponseFactory $responseFactory = null, StreamFactory $streamFactory = null)
-    {
+    /**
+     * @param Artax\Client    $client          HTTP client implementation.
+     * @param ResponseFactory $responseFactory Response factory to use or `null` to attempt auto-discovery.
+     * @param StreamFactory   $streamFactory   This parameter will be ignored and removed in the next major version.
```

Comment:
I think you can remove it directly withtout waiting for a next major and tag the next release a 0.2.0, it will be compliant with semver and as it's a fresh library it should be fine.